MekHQ/src/mekhq/gui/sorter/ScenarioStatusComparator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2020 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.gui.sorter;
+
+import mekhq.campaign.mission.Scenario;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+public class ScenarioStatusComparator implements Comparator<String>, Serializable {
+    private static final long serialVersionUID = -6287998488809978029L;
+
+    @Override
+    public int compare(String o1, String o2) {
+        int a = -1;
+        int b = -1;
+
+        // First we need to determine the numbers based on the name
+        for (int i = 0; i < Scenario.S_NUM; i++) {
```

Comment:
I think it would be more natural to have a helper method that converted the string to its sort order integer, then just compare the two integers.